### PR TITLE
Add generic scripter

### PR DESCRIPTION
**Pull request by martijnjanssen**

### CODE REVIEW

File: scripter/lua/lua.go [hunk 1]
```diff
@@ -65,32 +68,27 @@ func (l *luaScripter) Init(service string) error {
...
-	if sConn, ok = l.connections[ip]; !ok {
-		sConn = &luaConn{}
-		sConn.conn = conn
-		sConn.scripts = map[string]map[string]*lua.LState{}
-		sConn.abTester = l.abTester
+	ip := getConnIP(conn)
+
+	sConn, ok := l.connections[ip]
+	if !ok {
+		sConn = &luaConn{conn, map[string]map[string]*lua.LState{}, l.abTester}
```
Comment (Icety) [inline scripter/lua/lua.go]: Fine change, but use keys when applying variables this way, else everything breaks when the order changes in luaConn.

File: scripter/lua/lua.go [hunk 2]
```diff
@@ -101,3 +99,23 @@ func (l *luaScripter) GetConnection(service string, conn net.Conn) scripter.Conn
 	return &scripter.ConnectionStruct{Service: service, MyConn: sConn}
 }
 
+// Check whether scripter can handle incoming connection for the peeked message
+// Returns true if there is one script able to handle the connection
+func (l *luaScripter) CanHandle(service string, pMessage string) bool {
+	for _, ls := range l.canHandleStates[service] {
+		canHandle, err := callCanHandle(ls, pMessage)
```
Comment (Icety) [inline scripter/lua/lua.go]: pMessage is called message everywhere else, maybe use it here as 'message' as well?

**Review by Icety — COMMENTED** [review]
Looks good and works well, only needs commenting.

File: scripter/lua/lua.go (hunk 1 above)
Comment (martijnjanssen) [reply scripter/lua/lua.go]: I honestly don't think it's a problem that the code breaks when changing the underlying type.

File: scripter/lua/connection.go [hunk 3]
```diff
@@ -102,39 +102,62 @@ func (c *luaConn) AddScripts(service string, scripts map[string]string) {
...
+		Content: ls.ToString(-1),
 	}
 
 	ls.Pop(1)
 
 	return result, nil
 }
 
-func (c *luaConn) HandleScripts(service string, message string) (*scripter.Result, error) {
+func (c *luaConn) Handle(service string, message string) (*scripter.Result, error) {
```
Comment (Icety) [inline scripter/lua/connection.go]: Missing comments :(

File: services/services.go [hunk 4]
```diff
@@ -113,12 +113,12 @@ func WithConfig(c toml.Primitive) ServicerFunc {
 	}
 }
 
-func WithScripter(scr scripter.Scripter) ServicerFunc {
```
Comment (nordinvannes) [inline services/services.go:116]: Missing comments

**Review by nordinvannes — CHANGES_REQUESTED** [review]
After comments, it is ok!

File: services/services.go (hunk 4 above)
Comment (nordinvannes) [reply services/services.go:116]: Proven that comments do not matter in this file. Will be requested in new issue.

**Review by nordinvannes — APPROVED** [review]
LGTM! After comment.

**Pull request merged**